static/src/javascripts/projects/common/modules/spacefinder.js

```diff
@@ -0,0 +1,392 @@
+        : new Promise(resolve => {
+              let loadedCount = 0;
+              bean.on(rules.body, 'load', notLoaded, function onImgLoaded() {
+                  loadedCount += 1;
+                  if (loadedCount === notLoaded.length) {
+                      bean.off(rules.body, 'load', onImgLoaded);
+                      notLoaded.length = 0;
+                      resolve();
+                  }
+              });
+          });
+}, getFuncId);
+
+const onRichLinksUpgraded = memoize(
+    (rules: SpacefinderRules): Promise<void> =>
+        qwery('.element-rich-link--not-upgraded', rules.body).length === 0
+            ? Promise.resolve()
+            : new Promise(resolve => {
+                  mediator.once('rich-link:loaded', resolve);
+              }),
+    getFuncId
+);
+
+const onInteractivesLoaded = memoize((rules: SpacefinderRules): Promise<
+    void
+> => {
+    const notLoaded: Element[] = qwery(
+        '.element-interactive',
+        rules.body
+    ).filter((interactive: Element): boolean => {
+        const iframe: HTMLIFrameElement[] = ([...interactive.children].filter(
+            isIframe
+        ): any[]);
+        return !(iframe.length && isIframeLoaded(iframe[0]));
+    });
+
+    return notLoaded.length === 0 || !('MutationObserver' in window)
+        ? Promise.resolve()
+        : Promise.all(
+              notLoaded.map(
+                  (interactive: Element): Promise<void> =>
+                      new Promise(resolve => {
+                          new MutationObserver(
+                              (
+                                  records: MutationRecord[],
+                                  instance: MutationObserver
+                              ): void => {
+                                  if (
+                                      !records.length ||
+                                      !records[0].addedNodes.length ||
+                                      !isIframe((records[0].addedNodes[0]: any))
+                                  ) {
+                                      return;
+                                  }
+
+                                  const iframe = records[0].addedNodes[0];
+                                  if (isIframeLoaded((iframe: any))) {
+                                      instance.disconnect();
+                                      resolve();
+                                  } else {
+                                      iframe.addEventListener('load', () => {
+                                          instance.disconnect();
+                                          resolve();
+                                      });
+                                  }
+                              }
+                          ).observe(interactive, {
+                              childList: true,
+                          });
+                      })
+              )
+          ).then(() => undefined);
+}, getFuncId);
+
+const onAdsLoaded = memoize(
```

Comment:
I guess as a reader, I'm more interested in the signature of `onAdsLoaded`, rather than inferring it as the signature of the function inside `memoize`. So a small preference to outer.